e2e/rescheduling/input/reschedule_success.hcl

```diff
@@ -0,0 +1,24 @@
+job "test3" {
+  datacenters = ["dc1"]
+  type = "service"
+
+  group "t3" {
+    count = 3
+    task "t3" {
+      driver = "raw_exec"
+      config {
+         command = "bash"
+         args    = ["-c", "a=`cksum <<< \"${NOMAD_ALLOC_ID}\"| cut -d ' ' -f1`; if ! (( a % 2 )); then sleep 5000; else exit -1; fi"]
```

Comment:
1) I agree this is probably being too clever right now. Will think of a better way. 

2) I don't know much about the mock driver but open to changing all these tests to use mock driver in the future

3) I wanted to make some allocs fail and then subsequently succeed. Using the alloc index makes it so that allocs that fail, continue to fail and allocs that succeed do so the first time. That is a subtly different test scenario.